src/Features/LanguageServer/Protocol/Handler/Diagnostics/AbstractPullDiagnosticHandler.cs

```diff
@@ -215,12 +238,16 @@ private static VSDiagnostic ConvertDiagnostic(Document document, SourceText text
             Contract.ThrowIfNull(diagnosticData.DataLocation, $"Got a document diagnostic that did not have a {nameof(diagnosticData.DataLocation)}");
 
             var project = document.Project;
+
+            // Razor wants to handle all span mapping themselves.  So if we are in razor, return the raw doc spans, and
+            // do not map them.
+            var useMappedSpan = !IsRazorDocument(document);
```

Comment:
If Razor is the only documents that have span mapping, and they don't want span mapping of diagnostics, can we not just say that pull diagnostics never maps spans?

Reply:
No.  Pull diagnositcs should *def* map spans for the *non* razor cases.  For example, if we use winforms and they spit out `#line` directives, then we'd want the errors we report then to be properly mapped as per the file/line info that winforms asks for.  It seems like it's only razor that wants to intercept errors and handle the mapping themselves.

Reply:
I'm confused. Is this not the same span mapping as `ISpanMappingService` provides? I was going off the "Only razor docs have an ISpanMappingService" comment.

Reply:
Discussed offline.  This is confusing :)   There are two 'map the spans' concepts going on here.  One is the low-level concept that we support directly in the language with `#line`.  the other is the external razor concept where they track and map things.  in this case we're saying "if in razor, just don't do any sort of interpretation".  But if we're not in razor, we do want to do the normal C# #line mapping.